File: go-kata/internal/trees/binary_tree/post_order_traversal.go
package binary_tree

func walkPostOrder[T any](curr *BinaryNode[T], path *[]T) []T {
	if curr == nil {
		return *path
	}

	// Pre

	// Recursion
	walkPostOrder(curr.Left, path)
	walkPostOrder(curr.Right, path)

	// Post
	*path = append(*path, curr.Val)
	return *path
}

func PostOrderTraversal[T any](head *BinaryNode[T]) []T {
	return walkPostOrder(head, &[]T{})
}

File: typescript-kata/src/trees/postOrderTraversal.ts
import { BinaryNode } from "./node";

function walk(curr: BinaryNode<number> | undefined, path: number[]): number[] {
    if (!curr) {
        return path
    }

    // Pre
    //
    // Recursive
    walk(curr.left, path)
    walk(curr.right, path)

    // Post
    path.push(curr.val)
    return path
}

export default function PostOrderTraversal(head: BinaryNode<number>): number[] {
    return walk(head, [])
}
